Assets/Scripts/StandardSamples/UI/Common/UIDebugLog.cs

```diff
@@ -366,5 +386,33 @@ private void ScrollToBottom()
         {
             ScrollRect.verticalNormalizedPosition = 0;
         }
+
+        /// <summary>
+        /// Iterates over <see cref="logCacheList"/> and returns the total length of the string.
+        /// </summary>
+        /// <param name="numberOfMessagesToDequeueToReachThreshold">
+        /// The amount of messages that need to be dequeued before the
+        /// total length of the string is under <see cref="MaximumLogStringLength"/>.
+        /// </param>
+        /// <returns>The current length of the text string.</returns>
+        private int GetLengthOfLogCache(out int numberOfMessagesToDequeueToReachThreshold)
+        {
+            int totalLength = 0;
+            numberOfMessagesToDequeueToReachThreshold = 0;
+
+            foreach (string message in logCacheList)
+            {
+                int nextLength = message.Length;
+
+                if (totalLength + nextLength > MaximumLogStringLength)
+                {
+                    numberOfMessagesToDequeueToReachThreshold++;
+                }
+
+                totalLength += nextLength;
+            }
+
+            return totalLength;
+        }
```

Comment:
Instead of calculating the length of the log cache every time a log is added (which has a complexity of `O(n)`), we could store the total log length within a private `uint` value `_logCacheLength`. Each time a log is added, we would increment that field member with the length of the new log.

This would make the complexity of adding to the log `O(1)`. 

We would also need to make sure that when (and if) an item is removed from the queue, the `_logCacheLength` field member is decremented by the length of the log removed.

Reply:
I've updated the implementation per our discussions, and with permission from @WispyMouse. Marking this conversation as resolved and changing my review to approval.